### PR TITLE
Add left and right quotient operators

### PR DESCRIPTION
**Pull request by eliotwrobson**
Adds left and right quotient operators for NFAs with tests. Tests are a little basic, and the construction is a little rough around the edges, but overall should be solid. @caleb531 @Tagl

### CODE REVIEW

File: automata/fa/nfa.py [hunk 1]
```diff
@@ -510,6 +510,135 @@ def intersection(self, other):
...
+        """
+        Given two NFAs, M1 and M2, which accept the languages
+        L1 and L2 respectively, returns an NFA which accepts
+        the right quotient of L1 with respect to L2 (L1 / L2).
+
+        Construction is based off of the one described here:
+        https://cs.stackexchange.com/a/102043
+        """
+
+        # First, eliminita lambdas (cause problems with this algorithm)
```
Comment (Tagl) [inline automata/fa/nfa.py:523]:
```suggestion
        # First, eliminate lambdas because they cause problems with this algorithm
```

**Comment by caleb531** [comment]
@eliotwrobson What's the current status of this PR?

**Comment by eliotwrobson** [comment]
@caleb531 left it as a draft because I wanted to limit myself to one open PR at a time. But tests are written and working as expected, so I was planning just to rebase and flip to open.

**Comment by eliotwrobson** [comment]
@caleb531 Rebased and should be good to go!

**Review by caleb531 — APPROVED** [review]
@eliotwrobson This looks great! Thank you!

**Pull request merged**